libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java

```diff
@@ -108,8 +110,9 @@ protected CloudAdapterBase(BotFrameworkAuthentication withBotFrameworkAuthentica
                 logger.info(String.format("Sending activity. ReplyToId: %s", activity.getReplyToId()));
 
                 if (activity.isType(ActivityTypes.DELAY)) {
-                    int delayMs = Integer.valueOf((String) activity.getValue()) != null
-                        ? (int) activity.getValue() : DEFAULT_MS_DELAY;
+                    int delayMs = activity.getValue() != null
+                        ? new Double(String.valueOf(activity.getValue())).intValue()
```

Comment:
Out of curiosity: Can't the value be directly parsed to `int` instead of parsing it as `double` and getting the integer-value?

Reply:
I'll be confirming this as it's really weird

Reply:
I improved it parsing the Object value as Number (because Long/Float/Int/Double are Numbers) and then getting the `intValue` of the number object.

The tests are passing

![image](https://user-images.githubusercontent.com/11904023/125650748-bc190c8c-e1f5-49e4-82fd-01de7c331458.png)